scripts/migrations/037_checkfor_project_todos_in_moderations.py

```diff
@@ -0,0 +1,152 @@
+
+couch = couchdb.Server(COUCHSERVER)
+db = couch[DBNAME]
+
+# ----------------------------------------
+# queries
+# ----------------------------------------
+
+# get all project moderations with todos
+all_project_moderations = {"selector": {"type": {"$eq": "moderation"},"documentType": {"$eq": "PROJECT"},"$or": [{"projectAdditions": {"todos":{"$exists":True}}},{"projectDeletions": {"todos": {"$exists": True}}}]},"limit": 20000}
+
+# ---------------------------------------
+# functions
+# ---------------------------------------
+
+def checkEmptyTodosAnddeleteFromModeration(logFile, all_moderations):
+    log = {}
+    count = {}
+    count["total"] = str(len(all_moderations))
+    log['projectModerationRequestsHavingEmptyTodos'] = []
+    log['projectModerationRequestsHavingEmptyTodos'].append(count)
+    todo = "todos"
+    
+    for moderation in all_moderations:
+
+        projectAdditions = moderation.get("projectAdditions")
+        projectDeletions = moderation.get("projectDeletions")
+
+        todoslenghtinprojAddn = None
+        todoslenghtinprojDeln = None
+
+        if todo in projectAdditions:
+            todoslenghtinprojAddn = len(projectAdditions["todos"])
+
+        if todo in projectDeletions:
+            todoslenghtinprojDeln = len(projectDeletions["todos"])
+
+
+        if todoslenghtinprojAddn == 0 and todoslenghtinprojDeln == 0:
+            del(moderation["projectAdditions"]["todos"])
+            del(moderation["projectDeletions"]["todos"])
+        elif todoslenghtinprojAddn == 0 and todoslenghtinprojDeln is None:
+            del(moderation["projectAdditions"]["todos"])
+        elif todoslenghtinprojAddn is None and todoslenghtinprojDeln == 0:
+            del(moderation["projectDeletions"]["todos"])
+
+        if not DRY_RUN:
+            db.save(moderation)
+
+
+        openModRequestList = {}
+        openModRequestList['id'] = moderation.get('_id')
+        log['projectModerationRequestsHavingEmptyTodos'].append(openModRequestList)
+
+    json.dump(log, logFile, indent = 4, sort_keys = True)
+
+def checkForFilledTodosInModeration(logFile, all_moderations):
+    log = {}
+    todo = "todos"
+    log["moderation_requests_with_todos_having_data"] = []
+    moderationList = []
+    mods = {}
+    for moderation in all_moderations:
+        projectAdditions = moderation.get("projectAdditions")
+        projectDeletions = moderation.get("projectDeletions")
+
+        todoslenghtinprojAddn = None
+        todoslenghtinprojDeln = None
+
+        if todo in projectAdditions:
+            todoslenghtinprojAddn = len(projectAdditions["todos"])
+
+        if todo in projectDeletions:
+            todoslenghtinprojDeln = len(projectDeletions["todos"])
+
+        if (todoslenghtinprojAddn is not None and todoslenghtinprojAddn > 0) or (todoslenghtinprojDeln is not None and todoslenghtinprojDeln > 0):
+            moderationList.append(moderation.get("_id"))
+
+    if len(moderationList) > 0:
+        for x in moderationList:
```

Comment:
It would be good to have valid variable name in place of x

Reply:
refactored